packages/wrangler/src/d1/execute.tsx

```diff
@@ -135,7 +150,7 @@ export const Handler = withConfig<HandlerOptions>(
 			local,
 			config,
 			database,
-			isInteractive && !yes,
+			!json && isInteractive && !yes,
```

Comment:
also i'm 50/50 between piggy-backing on `shouldPrompt` vs just passing `--json` down to the exec functions and logging nothing when `json` is true

Reply:
I vote the latter, now we have an explicit flag we should keep being explicit.